shader_test.go

```diff
@@ -87,6 +87,76 @@ func Fragment(position vec4, texCoord vec2, color vec4) vec4 {
 	}
 }
 
+// Regression test for https://github.com/hajimehoshi/ebiten/issues/2525.
+func TestShaderWithDrawImageDoesNotWreckTextureUnits(t *testing.T) {
+	const w, h = 16, 16
+	rect := image.Rectangle{Max: image.Point{X: w, Y: h}}
+
+	dst := ebiten.NewImageWithOptions(rect, &ebiten.NewImageOptions{Unmanaged: true})
+	s, err := ebiten.NewShader([]byte(`package main
+
+func Fragment(position vec4, texCoord vec2, color vec4) vec4 {
+	return imageSrc0At(texCoord)
+}
+`))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	src1 := ebiten.NewImageWithOptions(rect, &ebiten.NewImageOptions{Unmanaged: true})
+	src1.Fill(color.RGBA{25, 0xff, 25, 0xff})
+	src2 := ebiten.NewImageWithOptions(rect, &ebiten.NewImageOptions{Unmanaged: true})
+	src2.Fill(color.RGBA{0xff, 0, 0, 0xff})
```

Comment:
`src0` and `src1` are better to align with the indices.

Reply:
Done.